ssl/s3_enc.c

```diff
@@ -415,14 +415,12 @@ void ssl3_digest_master_key_set_params(const SSL_SESSION *session,
                                        OSSL_PARAM params[])
 {
     int n = 0;
-    int cmd = EVP_CTRL_SSL3_MASTER_SECRET;
-
-    params[n++] = OSSL_PARAM_construct_int(OSSL_DIGEST_PARAM_CMD, &cmd,
-                                           NULL);
-    params[n++] = OSSL_PARAM_construct_octet_ptr(OSSL_DIGEST_PARAM_MSG,
-                                                (void **)&session->master_key,
-                                                 session->master_key_length,
-                                                 NULL);
+
+    params[n++] =
+        OSSL_PARAM_construct_octet_string(OSSL_DIGEST_PARAM_SSL3_MS,
+                                          (void *)session->master_key,
```

Comment:
will this line work (its a **)?

Reply:
Uhmmmm....  `session->master_key` is a `unsigned char []`...

https://github.com/openssl/openssl/blob/3f91ede9aea70774d9b5d509bc76d484ebaff6aa/ssl/ssl_locl.h#L521